cpp/oneapi/dal/algo/jaccard/detail/select_kernel.hpp

```diff
@@ -23,46 +23,43 @@
-                                                const descriptor_base &descriptor,
-                                                const Topology &data,
-                                                void *result_ptr) = 0;
-    virtual ~backend_base() {}
-};
+template <typename Policy, typename Descriptor, typename Topology>
+struct backend_base {
+    using float_t = typename Descriptor::float_t;
+    using task_t = typename Descriptor::task_t;
+    using method_t = typename Descriptor::method_t;
 
-template <typename Policy, typename Float, typename Method, typename Topology>
-struct ONEDAL_EXPORT backend_default : public backend_base<Policy, Topology> {
-    virtual vertex_similarity_result operator()(const Policy &ctx,
-                                                const descriptor_base &descriptor,
-                                                const Topology &data,
-                                                void *result_ptr) {
-        return call_jaccard_default_kernel_general(descriptor, data, result_ptr);
-    }
-    virtual ~backend_default() {}
+    virtual vertex_similarity_result<task_t> operator()(const Policy& ctx,
+                                                        const Descriptor& descriptor,
+                                                        const Topology& t,
+                                                        caching_builder& result_builder) = 0;
+    virtual ~backend_base() = default;
 };
 
-template <typename Float, typename Method>
-struct backend_default<dal::detail::host_policy,
-                       Float,
-                       Method,
-                       dal::preview::detail::topology<std::int32_t>>
-        : public backend_base<dal::detail::host_policy,
-                              dal::preview::detail::topology<std::int32_t>> {
-    virtual vertex_similarity_result operator()(
-        const dal::detail::host_policy &ctx,
-        const descriptor_base &descriptor,
-        const dal::preview::detail::topology<std::int32_t> &data,
-        void *result_ptr);
-    virtual ~backend_default() {}
+template <typename Policy, typename Descriptor, typename Topology>
+struct backend_default : public backend_base<Policy, Descriptor, Topology> {
+    static_assert(dal::detail::is_one_of_v<Policy, dal::detail::host_policy>,
+                  "Host policy only is supported.");
+
+    using float_t = typename Descriptor::float_t;
+    using task_t = typename Descriptor::task_t;
+    using method_t = typename Descriptor::method_t;
+
+    virtual vertex_similarity_result<task_t> operator()(const Policy& ctx,
+                                                        const Descriptor& descriptor,
+                                                        const Topology& t,
+                                                        caching_builder& result_builder) {
+        return vertex_similarity_kernel_cpu<float_t, method_t, task_t, Topology>()(ctx,
```

Comment:
why do we have '_cpu' in the name?

Reply:
To update.

Reply:
NO. It is aligning with ML train_kernel_cpu. 
Indeed it is CPU kernel for vertex_similarity.